Bang/GameEvents/CardEffects/GatlingActionHandler.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Bang.Characters;
+using Bang.Characters.Visitors;
+using Bang.Game;
+using Bang.GameEvents.CardEffects.States;
+using Bang.Players;
+using Bang.PlayingCards;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bang.GameEvents.CardEffects
+{
+    internal class GatlingActionHandler : CardActionHandler
+    {
+        public override HandlerState ApplyEffect(Game.Gameplay gameplay, Player attackPlayer, BangGameCard card)
+        {
+            var victimStatesList = new Dictionary<Player, HandlerState>();
+
+            foreach (var victim in gameplay.Players.Where(p => p.Id != attackPlayer.Id))
+            {
+                victimStatesList.Add(victim, TryShootPlayer(victim, gameplay));
+            }
+
+            return new WaitingMissedCardsAfterGatlingState(victimStatesList, gameplay);
+        }
+
+        private HandlerState TryShootPlayer(Player victim, Game.Gameplay gameplay)
```

Comment:
This method duplicates BangCardHandler. It will be better replace it on 
`
var bangHandler = new BangCardHandler();
bangHandler.ApplyEffect(victim, gamePlay, null); // null or gatling card
`

Reply:
Reasonably, Done